src/core/chmstate_mod.F90

```diff
@@ -25,30 +33,130 @@ module ChmState_Mod
-      INTEGER           :: NSpeicesSeaSalt
-      INTEGER, POINTER  :: AeroIndex(:)
-      INTEGER, POINTER  :: GasIndex(:)
-      INTEGER, POINTER  :: DustIndex(:)
-      INTEGER, POINTER  :: SeaSaltIndex(:)
+      INTEGER           :: nSpecies          ! Total Number of Species
+      INTEGER           :: nSpeciesGas       ! Number of Gas Species
+      INTEGER           :: nSpeciesAero      ! Number of Aerosol Species
+      INTEGER           :: nSpeciesDust      ! Number of Dust Species
+      INTEGER           :: NSpeicesSeaSalt   ! Number of SeaSalt Species
+      Integer, POINTER  :: SpeciesIndex(:)   ! Total Species Index
+      INTEGER, POINTER  :: AeroIndex(:)      ! Aerosol Species Index
+      INTEGER, POINTER  :: GasIndex(:)       ! Gas Species Index
+      INTEGER, POINTER  :: DustIndex(:)      ! Dust Species Index
+      INTEGER, POINTER  :: SeaSaltIndex(:)   ! SeaSalt Species Index
 
       !---------------------------------------------------------------------
       ! Reals
       !---------------------------------------------------------------------
       REAL(fp), POINTER :: ChmSpecies(:,:,:,:)
 
+      ! TODO: Add properties for species
    end type ChmStateType
 
 CONTAINS
    !
    !=========================================================================
    !
-   subroutine Chm_Allocate(Config_Opt, State_Grid, State_Chm, RC)
+   subroutine Chm_Allocate(Config_Opt, GridState, State_Species,State_Chm, RC)
 
       ! USES
       USE Config_Opt_Mod, ONLY : OptConfig
       USE GridState_Mod,  ONLY : GridStateType
-      USE ChmState_Mod,   ONLY : ChmStateType
+      USE Species_Mod,    Only : SpeciesType
+
+      IMPLICIT NONE
+
+      ! INOUT Params
+      type(OptConfig),     INTENT(in)    :: Config_Opt    ! Input Options object
+      type(GridStateType), INTENT(in)    :: GridState    ! Grid State object
+      type(ChmStateType),  INTENT(inout) :: State_Chm     ! Chm State object
+      type(SpeciesType),   POINTER       :: State_Species ! Species object
+      ! OUTPUT Params
+      INTEGER,             INTENT(OUT)   :: RC            ! Success or failure
+
+      ! Error handling
+      CHARACTER(LEN=255) :: ErrMsg
+      CHARACTER(LEN=255) :: thisLoc
+
+      ! Initialize
+      RC = CC_SUCCESS
+      ErrMsg = ''
+      thisLoc = ' -> at Chm_Allocate (in core/chmstate_mod.F90)'
+
+      ! Nullify all fields for safety's sake before allocating them
+      ! This can prevent compilation errors caused by uninitialized values
+      ! Nullify all fields for safety's sake before allocating them
+      ! This can prevent compilation errors caused by uninitialized values
+      State_Chm%ChmSpecies => NULL()
+
+      ! Allocate
+      ALLOCATE( State_Chm%ChmSpecies( GridState%NX, GridState%NY, &
+         GridState%number_of_levels, State_Chm%nSpecies ), &
+         STAT=RC )
+      CALL CC_CheckVar( 'ChmState%ChmSpecies', 0, RC )
+      IF ( RC /= CC_SUCCESS ) RETURN
+      State_Chm%ChmSpecies = 1e-31_fp
```

Comment:
Don't you have prescribed unset vals somewhere?

Reply:
There is something there but we should probably add one for `tiny` values rather than straight zero.  Might crash something in these chemical fields if they are set strictly to zero

Reply:
added a `TINY` type to precision mod